new_dawn_server/locations/api/resources.py

```diff
@@ -8,5 +12,71 @@ class Meta:
         always_return_data = True
         authorization = Authorization()
         allowed_methods = ["get", "post"]
+        filtering = {
+            'country': "exact",
+            'state': "exact"
+        },
         queryset = CityPreference.objects.all()
         resource_name = "city_preference"
+
+    def prepend_urls(self):
+        return [
+            url(r"^city_preference/get_country_list/$", self.wrap_view("get_country_list"),
+                name="api_get_country_list"),
+            url(r"^city_preference/get_state_for_country/$", self.wrap_view("get_state_for_country"),
+                name="api_get_state_for_country"),
+            url(r"^city_preference/get_city_for_state/$", self.wrap_view("get_city_for_state"),
+                name="api_get_city_for_country"),
+        ]
+
+    def get_country_list(self, request, **kwargs):
+        # TODO: Complete country list
+        self.method_check(request, allowed=["get"])
+        return self.create_response(request,
+            ClientResponse(
+                success=True,
+                message="Country list sent",
+                country_list=list(country_list.keys())
+            ).get_response_as_dict()
+        )
+
+    def get_state_for_country(self, request, **kwargs):
+        self.method_check(request, allowed=["get"])
+        data = self.deserialize(request, request.body, format=request.META.get("CONTENT_TYPE", "application/json"))
```

Comment:
You cannot deserialize request body on GET request since they don't have body. The GET request gets params directly from url. In django you should do:       

country = request.GET.get("country", "")